maintenance-filter-server/src/main/java/org/ccci/maintenance/util/MaintenanceWindowDaoFactory.java

```diff
@@ -0,0 +1,79 @@
+package org.ccci.maintenance.util;
+
+import org.ccci.maintenance.DatabaseMigrator;
+import org.ccci.maintenance.DatasourceManager;
+
+import javax.sql.DataSource;
+
+/**
+ * @author Matt Drees
+ */
+public class MaintenanceWindowDaoFactory
+{
+
+    String infinispanLocationParamName = "org.ccci.maintenance.window.infinispan.cache.location";
+
+    private final ConfigReader configReader;
+    private DatasourceManager datasourceManager;
+    private DataSource dataSource;
+    private Object cacheContainer;
+
+    public MaintenanceWindowDaoFactory(
+        ConfigReader configReader,
+        DatasourceManager datasourceManager)
+    {
+        this.configReader = configReader;
+        this.datasourceManager = datasourceManager;
+    }
+
+    public void initialize()
+    {
+        String location = configReader.getParameter(infinispanLocationParamName);
+        if (location == null)
+        {
+            dataSource = datasourceManager.lookupOrCreateDataSource();
+            initDatabaseIfNecessary(dataSource);
+        }
+        else
+        {
+            //avoid a hard dependency on infinispan, by not using Cache.class
+            Class<?> cacheContainerClass = getInfinispanCacheContainerClass();
+            cacheContainer = Lookups.doLookup(location, cacheContainerClass);
+        }
+    }
+
+    private Class<?> getInfinispanCacheContainerClass()
+    {
+        try
+        {
+            return Class.forName("org.infinispan.manager.CacheContainer");
+        }
+        catch (ClassNotFoundException e)
+        {
+            throw Exceptions.wrap(e);
```

Comment:
No, not really. `Exceptions` is a class in this project, and this does basically the same thing as `Throwables.propagate()`. I would have used that, except that I chose not to require guava for this. (I don't actually remember why I chose not to require guava, now that I think about it.)